### DISCUSSION TITLE
cannot select tabs with mouse with `--layout compact`

### DISCUSSION THREAD

**Discussion by tshepang** [zellij-org/zellij#1596] [Q&A] [answered]
Continuing from https://github.com/zellij-org/zellij/issues/1477#issuecomment-1188669976, the issue happens both on main and using v0.30.0.

I installed using `cargo install --path .` . `cargo make install` fails...
```
[cargo-make] INFO - Running Task: wasm-opt-plugins
[cargo-make] ERROR - Error while running duckscript: Source: Unknown Line: 8 - No such file or directory (os error 2)
[cargo-make] WARN - Build Failed.

**Comment by imsnif** [comment]
Could you try with `cargo make install /path/to/executable`?

**Reply by tshepang** [reply]
I get same error, with `cargo make install /usr/local/bin/zellij`.

**Reply by imsnif** [reply]
I think you need to install `binaryen` for the `wasm-opt` command. There are more details in CONTRIBUTING.md

**Reply by tshepang** [reply]
just installed `binaryen` (v108), and now...
```
❯ cargo make install /usr/local/bin/zellij
[cargo-make] INFO - cargo make 0.35.13
[cargo-make] INFO - Project: snode
[cargo-make] INFO - Build File: Makefile.toml
[cargo-make] INFO - Task: install
[cargo-make] INFO - Profile: development
[cargo-make] ERROR - Task install not found
[cargo-make] WARN - Build Failed.

**Reply by imsnif** [reply]
Which folder are you in?

**Reply by tshepang** [reply]
hehe, the wrong one :)

**Reply by tshepang** [reply]
I don't even remember which it was... looks like it's building now

**Reply by tshepang** [reply]
looks like it's doing the LTO thing, so taking forever... probably this must be limited to actual release builds

**Reply by imsnif** [reply]
Yeah, exactly. Otherwise it's a little bit slow for personal use. Out of curiosity: why would you like to run it from main and not the latest release?

**Reply by tshepang** [reply]
it's for this https://github.com/zellij-org/zellij/commit/50dd47c2e08f8ee513f9ac6f1b7854af93674ec9

**Reply by imsnif** [reply]
Aha, cool. Well, there's quite a bit of unreleased stuff in main and there might be some surprises. I'm planning on doing a thorough go-over before releasing sometime in the next week or so (hopefully sooner). So heads up.

**Reply by tshepang** [reply]
Note that v0.30 does not work either, so I did wonder if it was my setup somehow. My config is simple though...
```yaml
copy_clipboard: primary
scroll_buffer_size: 100000
pane_frames: false
scrollback_editor: emacsclient --tty
keybinds:
  unbind: [Ctrl: 'g']
  normal:
    - action: [SwitchToMode: Locked]
      key: [Esc]
  locked:
    - action: [SwitchToMode: Normal]
      key: [Esc]

**Reply by imsnif** [reply]
Do you mean with the compact layout and clicking the tabs? I just tried and for me the only thing I get is the offset bug in the original issue... or do you mean something else?

**Reply by tshepang** [reply]
ahh, I see the offset bug... I did not see it before because I was testing with only two tabs

**Reply by imsnif** [reply]
Cool, glad it's sorted :)

**Reply by tshepang** [reply]
hm, is it supposed to be fixed... it's still happenin here (on main)

**Reply by imsnif** [reply]
For me it works... Are you sure you're running the built executable? If not that, then it could be a matter of deleting the plugin cache (for me it's in ~/.cache/zellij) - but I'd be surprised if you got the plugins from v30 somehow...

**Reply by tshepang** [reply]
I did that, same problem
```
❯ zellij -V
zellij 0.31.0

❯ rm -rf ~/.cache/zellij

❯ which zellij
/usr/local/bin/zellij

**Reply by imsnif** [reply]
Hum. Bit of a head-scratcher. The only thing I can think of is that something went wrong with the build. Sorry for the trouble, but could you reclone, make sure the target directory is empty, then do a fresh `cargo make install /path`, delete the plugin cache and try again?

**Reply by tshepang** [reply]
did all that, same problem

**Reply by imsnif** [reply]
Hrm - I think I gave you the wrong plugin cache folder. It should be ~/.local/share/plugins (you can make sure by running `zellij setup --check`). Maybe if you delete that and re-run?

**Reply by imsnif** [reply]
You could also try running with `--data-dir /some/empty/folder` to skip the cache... might help troubleshoot this?

**Reply by tshepang** [reply]
heh, that fixed it! I really should have checked.

**Reply by imsnif** [reply]
Nah, no worries friend. I'm a little bit puzzled as to how this happened. The plugin cache is supposed to be versioned - so unless you compiled from main after the last version but before this got fixed, I'm not too sure how this happened. Any clues?

**Reply by tshepang** [reply]
That sounds like what could be what happened, because I started playin with zullij around the same time as that fix was merged.

**Reply by imsnif** [reply]
Ahaa, okay. Now it makes sense then, thanks for putting my mind at ease. :)
When we release a version, before upgrading I'd recommend you delete the cache again so you'll get the latest plugins.

**Reply by tshepang** [reply]
Will stay with released versions once v0.31 is out, and that should sort things out (since the cache is versioned). Maybe the versioning should be more fine-grained, where it changes when plugins change? Is that a practical thing?

**Reply by imsnif** [reply]
For sure, I think there was some work done in this direction at some point. For now though, we're set to replace this entire system because - as you see - it's causing us a lot of pain :) The new plugin system will use some scripting language and thus forgo the need to cache assets on the HD. So this whole problem goes away.

**Comment by tshepang** [comment, marked as answer]
https://github.com/zellij-org/zellij/discussions/1596#discussioncomment-3192542
